experimental/packages/opentelemetry-sdk-metrics-base/src/export/MetricReader.ts

```diff
@@ -15,51 +15,206 @@
+  protected abstract onInitialized(): void;
+
+  /**
+   * Handle a shutdown signal by the SDK.
+   *
+   * <p> For push exporters, this should shut down any intervals and close any open connections.
+   * @protected
+   */
+  protected abstract onShutdown(): Promise<void>;
+
+  /**
+   * Handle a force flush signal by the SDK.
+   *
+   * <p> In all scenarios metrics should be collected via {@link collect()}.
+   * <p> For push exporters, this should collect and report metrics.
+   * @protected
+   */
+  protected abstract onForceFlush(): Promise<void>;
+
+  /**
+   * Collect all metrics from the associated {@link MetricProducer}
+   */
+  collect(options: ReaderCollectionOptions): void {
+    const timeout = options.timeoutMillis ?? 10000;
+    const done = options.done ?? (_result => {
+    });
+
     if (this._metricProducer === undefined) {
-      throw new Error('MetricReader is not bound to a MeterProvider');
+      done({
+        code: ReaderResultCode.FAILED,
+        error: new Error('MetricReader is not bound to a MetricProducer'),
+      });
+      return;
     }
-    const metrics = await this._metricProducer.collect();
 
-    // errors thrown to caller
-    await this._exporter.export(metrics);
+    // Subsequent invocations to collect are not allowed. SDKs SHOULD return some failure for these calls.
+    if (this._shutdown) {
+      done({
+        code: ReaderResultCode.FAILED,
+        error: new Error('Collection is not allowed after shutdown'),
+      });
+      return;
+    }
+
+    promiseWithTimeout(this._metricProducer.collect(), timeout, done);
   }
 
-  async shutdown(): Promise<void> {
+  /**
+   * Shuts down the metric reader, the callback will fire after the specified timeout or after completion.
+   *
+   * <p> NOTE: this operation will continue even after the timeout fires the callback.
+   * @param options
+   */
+  shutdown(options: ReaderForceFlushOptions): void {
+    const timeout = options.timeoutMillis ?? 10000;
+    const done = options.done ?? (_result => {
+    });
+
+    // Do not call shutdown again if it has already been called.
     if (this._shutdown) {
+      done({
+        code: ReaderResultCode.FAILED,
+        error: new Error('Cannot call shutdown twice.')
+      });
       return;
     }
 
-    this._shutdown = true;
-    // errors thrown to caller
-    await this._exporter.shutdown();
+    promiseWithTimeout(this.onShutdown(), timeout, result => {
+        if (result.code === ReaderResultCode.SUCCESS) {
+          this._shutdown = true;
+        }
+        done(result);
+      }
+    );
+
   }
 
-  async forceFlush(): Promise<void> {
+  /**
+   * Flushes metrics read by this reader, the callback will fire after the specified timeout or after completion.
+   *
+   * <p> NOTE: this operation will continue even after the timeout fires the callback.
+   * @param options options with timeout (default: 10000ms) and a result callback.
+   */
+  forceFlush(options: ReaderShutdownOptions): void {
```

Comment:
Ditto, even BasicTracerProvider defines `forceFlush` and `shutdown` return value as `Promise<void>`.